common/metrics/client.go

```diff
@@ -125,3 +135,13 @@ func mergeMapToRight(src map[string]string, dest map[string]string) {
 		dest[k] = v
 	}
 }
+
+func stringifyScope(scopeIdx int, tags ...Tag) string {
+	var b strings.Builder
+	b.WriteString(string(scopeIdx))
```

Comment:
this looks like atleast as expensive (if not more) as creating a sub-scope itself. Try writing a benchmark test to compare allocations with the baseline